### PR TITLE
Add MovieLens Example, Triplet Precision

### PR DESCRIPTION
**Pull request by guillaumegenthial**
# Description

Add MovieLens example (notebook, module, configs).

Refactor examples into multiply / movielens

Minor fixes (weighted average, path, etc.)

Add TripletPrecision layer

## Type of change

- [ ] Feature
- [ ] Fix
- [ ] Release

## Checklist

- [ ] I have followed the [CONTRIBUTING](https://github.com/criteo/deepr/blob/master/docs/CONTRIBUTING.rst) guidelines.
- [ ] I have updated the [CHANGELOG](https://github.com/criteo/deepr/blob/master/docs/CHANGELOG.rst).

### CODE REVIEW

**Comment by guillaumegenthial** [comment]
precision@10 = 0.0874
recall@10 = 0.10513672443536348
f1_score@10 = 0.08110665024298455
precision@20 = 0.07973
recall@20 = 0.18622017598325177
f1_score@20 = 0.09623903174448499
precision@50 = 0.064152
recall@50 = 0.3455439984337652
f1_score@50 = 0.09742900473438484

**Comment by guillaumegenthial** [comment]
Scores of WAN with learning_rate = 0.001

INFO:deepr.examples.movielens.jobs.evaluate:precision@10 = 0.09053000000000001
recall@10 = 0.11012133008960352
f1_score@10 = 0.08430578703070332
INFO:deepr.examples.movielens.jobs.evaluate:precision@20 = 0.08503500000000001
recall@20 = 0.2000903002221937
f1_score@20 = 0.10297113876576268
INFO:deepr.examples.movielens.jobs.evaluate:precision@50 = 0.06888000000000001
recall@50 = 0.3718694075515831
f1_score@50 = 0.1047276619072677

**Comment by guillaumegenthial** [comment]
Results Transformer(100), 1 block, 4 heads, lr=0.001, 10 shuffles of timelines, using <next> movie

INFO:deepr.examples.movielens.jobs.predict:Wrote predictions to transformer_next_shuffle/predictions.parquet.snappy
INFO:deepr.examples.movielens.jobs.evaluate:precision@10 = 0.09195000000000002
recall@10 = 0.10847400715929184
f1_score@10 = 0.08451091429814592
INFO:deepr.examples.movielens.jobs.evaluate:precision@20 = 0.082245
recall@20 = 0.18724129340375895
f1_score@20 = 0.09841842204205073
INFO:deepr.examples.movielens.jobs.evaluate:precision@50 = 0.06594000000000001
recall@50 = 0.34630333300132266
f1_score@50 = 0.09956293818576487
